internal/handlers/pg/pgdb/transactions.go

```diff
@@ -90,20 +90,20 @@ func (pgPool *Pool) InTransaction(ctx context.Context, f func(pgx.Tx) error) (er
 // If f returns a transactionConflictError, the transaction is retried.
 // If after maxRetries the transaction still fails, the last error unwrapped from transactionConflictError is returned.
 func (pgPool *Pool) InTransactionRetry(ctx context.Context, f func(pgx.Tx) error) (err error) {
-	var tcErr *transactionConflictError
-
 	for retry := 0; retry < maxRetries; retry++ {
-		err := pgPool.InTransaction(ctx, f)
+		err = pgPool.InTransaction(ctx, f)
+
+		var tcErr *transactionConflictError
 
 		switch {
 		case err == nil:
 			return nil
 		case errors.As(err, &tcErr):
 			ctxutil.Sleep(ctx, delay)
 		default:
-			return err
+			return lazyerrors.Error(err)
```

Comment:
Added more `lazyErrors` to have more debug info, for example:

```
(InternalError) [transactions.go:108 pgdb.(*Pool).InTransactionRetry] [transactions.go:77 pgdb.(*Pool).InTransaction] [msg_create.go:97 pg.(*Handler).MsgCreate.func1] [collections.go:129 pgdb.CreateCollection] [settings.go:91 pgdb.upsertSettings] ERROR: deadlock detected (SQLSTATE 40P01)```

this is helpful as one can see the full chain of calls.